addon/-private/async-arrow-runtime.js

```diff
@@ -16,6 +16,13 @@ export function buildTask(contextFn, options, taskName, bufferPolicyName) {
 
   const result = contextFn();
 
+  if (optionsWithBufferPolicy && optionsWithBufferPolicy.waitFor) {
+    result.generator = optionsWithBufferPolicy.waitFor(result.generator);
+
+    optionsWithBufferPolicy = Object.assign({}, optionsWithBufferPolicy);
+    delete optionsWithBufferPolicy.waitFor;
+  }
+
```

Comment:
I think this probably should be moved into a modifier (_a la_ https://github.com/machty/ember-concurrency/issues/513), so that it works regardless of whether a task is running through the AyncArrowRuntime or not